src/datachain/lib/udf_signature.py

```diff
@@ -18,7 +18,7 @@ def __init__(self, chain: str, msg):
 @dataclass
 class UdfSignature:
     func: Union[Callable, UDFBase]
-    params: Sequence[str]
+    params: dict[str, Union[DataType, Any]]
```

Comment:
UDF signature params is now a dictionary with param names as keys and param types as values, instead of just a list of param names.